### PR TITLE
feat(wallet)!: apply FRC-0102 envelope to sign/verify by default

### LINKED ISSUE #6442: Implement FRC-102 in `forest-wallet` for consistency across Filecoin wallets
# Summary

Migrate https://github.com/filecoin-project/lotus/issues/13256 to Forest (`forest-wallet`). 

The implementation in Go is done is done so it could be used as a reference.

# Completion Criteria

- [ ] Implementation
- [ ] Tests
- [ ] Docs

# Additional Links & Resources

### PR DESCRIPTION
**Pull request by 0xDevNinja**
## Summary of changes

Reopens the work from #6967, which was closed for inactivity. Rebased on current `main` and the two outstanding review comments are addressed.

Changes introduced in this pull request:
- `forest-wallet sign` and `forest-wallet verify` now apply the FRC-0102 signing envelope (`0x19 || "Filecoin Signed Message:\n" || ascii(len(msg)) || msg`) to the message by default.
- A `--raw` flag on both sides reproduces the previous raw-bytes behaviour.
- Fixes the sign/verify payload-shape mismatch in remote mode (`wallet_sign` now takes `Vec<u8>` and passes the payload through unchanged, so both paths operate on the same bytes).

## Reference issue to close (if applicable)

Closes #6442

## Other information and links

Follow-up from the review on #6967 (thanks @akaladarshi for the guidance there). Addressed both remaining comments:
- Removed the now-unused `Filsnap` entry from `.config/forest.dic` (`interoperating` is still referenced in a doc comment, kept).
- Collapsed the near-identical `wrap_frc0102` unit tests into parametrized `#[rstest]` cases (envelope-shape cases and the length-boundary cases).

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation. All new code adheres to the team's documentation standards,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the CHANGELOG is up-to-date. All user-facing changes should be reflected in this document.

## Summary by CodeRabbit

* **New Features**
  * `forest-wallet sign` and `forest-wallet verify` now apply the **FRC-0102 signing envelope** to messages by default.
  * Use the new `--raw` flag on either command to preserve the previous raw-bytes message behavior.
  * Message/signature inputs and outputs remain **hex-based**, with signatures returned as hex.
* **Chores**
  * Updated the onboarded dictionary word list.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -49,6 +49,8 @@
 
 - [#7164](https://github.com/ChainSafe/forest/issues/7164): JSON-RPC authentication is now performed once per connection (e.g. at the WebSocket upgrade) instead of on every request, matching Lotus. Note that token expiry is no longer re-checked for the lifetime of an established connection.
 
+- [#6442](https://github.com/ChainSafe/forest/issues/6442): `forest-wallet sign` and `forest-wallet verify` now apply the FRC-0102 signing envelope to the message by default. Pass `--raw` on both sides to reproduce the previous raw-bytes behaviour.
```
Comment (LesnyRumcajs) [inline CHANGELOG.md]: Let's move that to the current release train

**Comment by LesnyRumcajs** [comment]
LGTM, just please update the changelog.

**Comment by 0xDevNinja** [comment]
Done — moved the entry up to the unreleased train. Thanks for the review!

**Review by LesnyRumcajs — APPROVED** [review]
LGTM, thanks!
